src/js/services/StorageService.js

```diff
@@ -0,0 +1,17 @@
+// This is the service to interact with the Chrome storage
+// We can switch this service to IndexedDBService if we want to use IndexedDB instead of Chrome storage
+export class StorageService {
+  async get(key, defaultValue = null) {
+    return new Promise((resolve) => {
+      chrome.storage.sync.get({ [key]: defaultValue }, (data) => {
+        resolve(data[key]);
```

Comment:
`if (chrome.runtime.lastError) {
            reject(new Error(`\`Error setting key "${key}": ${chrome.runtime.lastError.message}\``));
          } 
 else {
            resolve();
          }`
 Adding `chrome.runtime.lastError` checks can handle errors from the Chrome Storage API.